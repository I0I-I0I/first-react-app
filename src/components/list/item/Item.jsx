import styles from "./Item.module.css";
import Button from "../../ui/button/Button";

const Item = ({ item, remove }) => (
	<li key={item.objectIDkey} className={styles.item}>
		<a target="_blank" style={{ width: "40%" }} href={item.url}>
			{item.title}
		</a>
		<span style={{ width: "30%" }}>{item.author}</span>
		<span style={{ width: "10%" }}>{item.num_comments}</span>
		<span style={{ width: "10%" }}>{item.points}</span>
		<span style={{ width: "10%" }}>
			<Button onClick={() => remove(item)} size="small">
				delete
			</Button>
		</span>
	</li>
);

export default Item;
