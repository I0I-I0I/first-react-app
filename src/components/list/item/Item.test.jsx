import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import Item from "./Item";

const story = {
	title: "React",
	url: "https://reactjs.org/",
	author: "Jordan Walke",
	num_comments: 3,
	points: 4,
	objectID: 0,
};

describe("Item", () => {
	it("renders all properties", () => {
		render(<Item item={story} />);

		expect(screen.getByText("Jordan Walke")).toBeInTheDocument();
		expect(screen.getByText("React")).toHaveAttribute(
			"href",
			"https://reactjs.org/",
		);
	});

	it("renders a clickable dismiss button", () => {
		render(<Item item={story} />);
		expect(screen.getByRole("button")).toBeInTheDocument();
	});

	it("clicking the dismiss button calls the callback handler", () => {
		const handleRemoveItem = vi.fn();
		render(<Item item={story} remove={handleRemoveItem} />);
		const button = screen.getByRole("button");
		fireEvent.click(button);
		expect(handleRemoveItem).toHaveBeenCalledTimes(1);
	});

	it("renders snapshot", () => {
		const { container } = render(<Item item={story} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
