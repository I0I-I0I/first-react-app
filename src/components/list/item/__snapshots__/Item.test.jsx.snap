// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Item > renders snapshot 1`] = `
<li
  class="_item_d7d403"
>
  <a
    href="https://reactjs.org/"
    style="width: 40%;"
    target="_blank"
  >
    React
  </a>
  <span
    style="width: 30%;"
  >
    Jordan Walke
  </span>
  <span
    style="width: 10%;"
  >
    3
  </span>
  <span
    style="width: 10%;"
  >
    4
  </span>
  <span
    style="width: 10%;"
  >
    <button
      class="_button_3c5126 _buttonSmall_3c5126"
    >
      delete
    </button>
  </span>
</li>
`;
